tests/specs/controllers/OpenEAppController.test.js

```diff
@@ -321,4 +323,48 @@ describe('OpenEAppController', () => {
             expect(result).to.be.true;
         });
     });
+
+    describe('_hasApplicationExpired', () => {
+        it('calls getApplicationReceipt method from CasebookService to stream file', async () => {
+            // when
+            const getReceipt = sandbox
+                .stub(CasebookService, 'getApplicationReceipt')
+                .resolves({
+                    data: {
+                        pipe: () => {},
+                    },
+                });
+
+            sandbox.stub(HelperService, 'getUserData').callsFake(() => ({
+                loggedIn: true,
+            }));
+            await OpenEAppController.downloadReceipt(reqStub, resStub);
+
+            // then
+            expect(getReceipt.calledOnce).to.be.true;
+        });
+
+        it('triggers serverError when user is not logged in', async () => {
+            // when
+            sandbox.stub(HelperService, 'getUserData').callsFake(() => ({
+                loggedIn: false,
+            }));
+            await OpenEAppController.downloadReceipt(reqStub, resStub);
+
+            // then
+            expect(resStub.serverError.calledOnce).to.be.true;
+        });
+
+        it('triggers serverError if application ref is undefined', async () => {
+            // when
+            sandbox.stub(HelperService, 'getUserData').callsFake(() => ({
+                loggedIn: true,
+            }));
+            reqStub.params.applicationRef = undefined;
+            await OpenEAppController.downloadReceipt(reqStub, resStub);
+
+            // then
+            expect(resStub.serverError.calledOnce).to.be.true;
```

Comment:
is the error message `'Missing application reference'`?  

the error as it is, should only throw if `reqStub.params.applicationRef = 'undefined' //string containing the word undefined` not `undefined` (the keyword).

Reply:
This is down Good point. I will make that change 👍

Reply:
Yes, the error message is `'Missing application reference'`. I did originally think about testing the throw but I've only exposed one function and the other unexposed function is the one that actually does the throwing. So I'm not sure how to go about testing that without exposing it as well.

Reply:
that's ok if you're not "close" enough to a function that throws. I just wanted to make sure that test was throwing for the reason you were expecting (if that made any sense)